lib/client.js

```diff
@@ -422,10 +426,9 @@ class Client extends EventEmitter {
         return
       }
 
-      if (!body) {
-        // TODO: InvalidReturnValueError?
+      if (!body || typeof body.pipe !== 'function') {
         if (!ret.destroyed) {
-          ret.destroy(new InvalidArgumentError('invalid body'))
+          ret.destroy(new InvalidReturnValueError('invalid body'))
```

Comment:
I would write a more descriptive error message, as the user might not understand what's going on here.

Reply:
How about now?